File: Algorithm/9-palindrome-number.py
# 9. Palindrome Number

# Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward.

# Solution (Runtime: 440 ms, faster than 38.19% of Python3 submissions):

def isPalindrome(self, x):
	"""
	:type x: int
	:rtype: bool
	"""
	if x < 0:
		return False
	return x == int(str(x)[::-1])

File: Algorithm/9-palindrome-number.java
// 9. Palindrome number
// 75 ms, faster than 97.04% of java submissions
// 29.2 MB, less than 32.24% java sunmissions

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int last = x % 10;
        int reduce = x / 10;
        int reverse = 0;
        while (reduce != 0) {
            reverse += last;
            last = reduce % 10;
            reduce = reduce / 10;
            reverse *= 10;
        }
        reverse += last;
        
        return x == reverse;
    }
}